conda/version.py

```diff
@@ -302,6 +302,10 @@ def __new__(cls, spec):
         self.spec = spec
         if isinstance(spec, tuple):
             self.match = self.all_match_ if spec[0] == 'all' else self.any_match_
+        elif spec.startswith("^") and spec.endswith("$"):
```

Comment:
Moreover I really don't care either way. Just figured there was a way to give you both exactly what you were asking for. Then I read the rest of the conversation and realized you already agreed on something.

The PR is fine to me as is, or we can make it more robust as I suggested above. Completely indifferent.